WPMedia/engine-theme-sdk
# Tmedia 264 localize date passed in value

## Description
Pass in any locale

## Jira Ticket
- [TMEDIA-264](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-264&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
A client can specify a currently-unsupported dateLocalization.language (eg pt_PT).  Currently, for example, "EN" maps to "en_US" (American "Simplified" English). 

Any of the existing dateLocalization.language mappings (e.g. how we map “en” to “en_US”) should still be in effect 

If a customer adds a new value for dateLocalization.language that we don’t have mapped to something, then we should use that value directly 

Examples: en_GB should use en_GB

## Test Steps
- See doc debugging postinstall


// To test this, see debugging-timezone-postinstall-dates.md
// 1. Add a blocks.json file in the src/ folder from above doc
// 2. run `rm -rf node_modules && npm ci --production`
// 3. run `ls node_modules/timezone`
// 4. See pt_PT in the list
// 5. You can uncomment this test

// eslint-disable-next-line max-len
// it('support portuguese in portugal language and portugal lisbon timezone when setup with blocks.json', () => {
//   expect(localizeDateHelper(
//     '2000-01-02 01:00',
//     '%B %d, %Y %l:%M %P %Z',
//     'pt_PT',
//     'Europe/Lisbon',
//   )).toMatchInlineSnapshot('"Janeiro 02, 2000  1:00  WET"');
// });

## Effect Of Changes
### Before
- Portuguese and other locales not allowed

### After
- Can pass in any locale

## Dependencies or Side Effects

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added. -> yas
